gluten-core/src/main/scala/org/apache/spark/util/SparkDirectoryUtil.scala

```diff
@@ -67,29 +68,31 @@ class SparkDirectoryUtil private (val roots: Array[String]) extends Logging {
 }
 
 object SparkDirectoryUtil extends Logging {
-  private var INSTANCE: SparkDirectoryUtil = _
+  private val INSTANCE = new AtomicReference[SparkDirectoryUtil]()
 
-  def init(conf: SparkConf): Unit = synchronized {
+  def init(conf: SparkConf): Unit = {
     val roots = Utils.getConfiguredLocalDirs(conf)
     init(roots)
   }
 
-  private def init(roots: Array[String]): Unit = synchronized {
-    if (INSTANCE == null) {
-      INSTANCE = new SparkDirectoryUtil(roots)
+  private def init(roots: Array[String]): Unit = {
+    if (INSTANCE.get() == null) {
+      INSTANCE.compareAndSet(null, new SparkDirectoryUtil(roots))
```

Comment:
The code is workable but the constructor of `SparkDirectoryUtil` may be unnecessarily called in concurrent invocations.

We can use `AtomicBoolean` instead if unable to avoid this.

Reply:
For example,  thread 1 and thread 2 can both reach line 80 at the same time. Although only one thread will initialize the `INSTANCE`, another thread will still do `new SparkDirectoryUtil(roots)` which is not necessary.